app/javascript/listings/singleListing/SingleListing.jsx

```diff
@@ -32,85 +32,42 @@ export class SingleListing extends Component {
     );
   };
 
-  listingInline = (
-    listing,
-    currentUserId,
-    onChangeCategory,
-    onOpenModal,
-    onAddTag,
-  ) => {
+  const listingInline = (props) => {
+    const { listing } = props;
     return (
       <div
         className="single-listing relative crayons-card"
         id={`single-listing-${listing.id}`}
         data-testid={`single-listing-${listing.id}`}
       >
-        <div className="listing-content p-4">
-          {this.listingContent(
-            listing,
-            currentUserId,
-            onChangeCategory,
-            onOpenModal,
-            onAddTag,
-          )}
-        </div>
+        <div className="listing-content p-4">{listingContent(props)}</div>
       </div>
     );
   };
 
-  listingModal = (
-    listing,
-    currentUserId,
-    onChangeCategory,
-    onOpenModal,
-    onAddTag,
-  ) => {
+  const listingModal = (props) => {
```

Comment:
I feel like these functions should be separate Preact components e.g. 
```suggestion
  const ListingModal = (props) => {
```

and then we could reference them below like `<ListingModal { ...props } />`. But it's a little outside the scope of this refactor, so I'm happy to approve and let you decide if it merits a follow up 😄